src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,376 @@
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "Name, hostList and allowedUserGroups should be "
+                    + "specified for successful partition update");
+        }
+
+        final Context context = createContextWithUpdatePartitionName(updateName);
+        final CommandResult sinfoResult = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIfExecutionResultIsEmpty(sinfoResult);
+
+        final SlurmQueue partitionData = getPartitionData(sinfoResult.getStdOut().get(0));
+
+        final List<String> currentNodesDecrypted = partitionData.getNodelist();
+        final List<String> userGroups = partitionData.getGroups();
+        final int cpus = partitionData.getCpus();
+
+        final List<String> updateHostListDecrypted = decryptGroupOfNodes(updateHostList);
+        sortList(updateUserGroups);
+
+        if (updateHostListDecrypted.equals(currentNodesDecrypted) && updateUserGroups.equals(userGroups)) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "New partition properties and the current one are "
+                    + "equal");
+        }
+        fillContextWithDataToUpdate(context, updateUserGroups, updateHostListDecrypted);
+
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        final Map<String, Integer> slotDescription = getNodesDescription(updateHostListDecrypted,
+                CPUS_PER_NODE_BY_DEFAULT);
+
+        return Queue.builder()
+                .name(updateName)
+                .hostList(updateHostListDecrypted)
+                .allowedUserGroups(updateUserGroups)
+                .slots(new SlotsDescription(cpus, slotDescription))
+                .build();
+    }
+
+    @Override
+    public List<Queue> listQueues() {
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, new Context()));
+        checkIsResultIsCorrect(result);
+
+        if (result.getStdOut().isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        return fillQueueNameFromOutput(result.getStdOut());
+    }
+
+    @Override
+    public List<Queue> listQueues(final QueueFilter queueFilter) {
+        final Context context = new Context();
+        if (queueFilter.getQueues() != null) {
+            context.setVariable(SCONTROL_PARTITION_NAME, queueFilter.getQueues());
+        }
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        if (result.getStdOut().isEmpty()) {
+            return Collections.emptyList();
+        }
```

Comment:
we guess we could remove these rows at all

Reply:
fixed